docs/TRACEZ_DESIGN.md

```diff
@@ -0,0 +1,175 @@
+# OpenTelemetry SDK Contrib - /tracez and /traceconfigz Design Doc
+
+This file contains information about the design choices for the OpenTelemetry /tracez and
+/traceconfigz zPages.
+
+## Introduction
+
+The OpenTelemetry zPages are a set of dynamically generated HTML pages that display trace and
+metrics data for a running process. Within this set, the /tracez zPage is a web page that displays
+trace information about running spans, sample span latencies, and sample error spans. On the other
+hand, the /traceconfigz zPage is a web page that allows the users to change tracing parameters, such
+as sampling probability and max number of attributes.
+
+### /tracez zPage
+
+The /tracez zPage displays information on running spans, sample span latencies, and sample error
+spans. The data is aggregated into a summary-level table:
+
+![tracez-table](img/tracez-table.png)
+
+In addition, you can click on each of the counts in the table cells to access the corresponding span
+details. For example, here are the details of the `ChildSpan` latency sample (row 1, col 4):
+
+![tracez-details](img/tracez-details.png)
+
+### /traceconfigz zPage
+
+The /traceconfigz zPage displays information about the currently active tracing configuration and 
+provides an interface for users to modify relevant parameters:
+
+![traceconfigz](img/traceconfigz.png)
+
+## Motivation
+
+At the moment, there are no zPages for the Java implementation of OpenTelemetry. The goal of this
+project, as suggested by the title, is to create the TraceZ zPage for the Java SDK. The motivation
+for this stems from the usefulness of zPages. Because of how they aggregate and display data, zPages
+are great development tools for OpenTelemetry instrumentation and lightweight monitoring. In
+addition, the implementation of the Java zPages can pave the way for future implementations in other
+languages, such as Python, which have yet to be built.
+
+## Proposed Design
+
+We have divided the implementation of the /tracez and /traceconfigz zPages into two frontend-centric
```

Comment:
Not sure if this qualifier is necessary. I think the frontend + backend section are self-explanatory.

Reply:
That is we can delete this Proposed Design section right?

Reply:
I'll remove it.